src/Libraries/RevitNodes/GeometryConversion/CurveUtils.cs

```diff
@@ -124,6 +149,72 @@ private static bool IsLineLikeInternal(IList<XYZ> points)
             }
         }
 
+        // This test checks end points with directionality. Curves that share end points
+        /// but are reversed are not considered similar.
+        private static bool CurvesHaveSameEndpoints(Curve a, Curve b, double tolerance)
+        {
+            return a.GetEndPoint(0).IsAlmostEqualTo(b.GetEndPoint(0), tolerance) &&
+                a.GetEndPoint(1).IsAlmostEqualTo(b.GetEndPoint(1));
+        }
+
+        private static bool AreSimilar(Line a, Line b, double tolerance)
+        {
+            return CurvesHaveSameEndpoints(a,b, tolerance);
+        }
+
+        private static bool AreSimilar(Arc a, Arc b, double tolerance)
+        {
+            return a.Center.IsAlmostEqualTo(b.Center) && 
+                a.Radius.AlmostEquals(b.Radius, 1.0e-6) && 
+                a.IsCyclic == b.IsCyclic;
```

Comment:
The start point and end point are already compared in an outer check. I agree we should add a normal check.